isp/convenience.go

```diff
@@ -246,5 +247,16 @@ func NewWithAuthHeader(header string) *HighLevelClient {
 // TODO: remove this... for some reason it is missing in the generated output.
 // At some point we need to switch to a less wonky code generator.
 func parameterToString(v interface{}, s string) string {
-	return fmt.Sprintf("%v", v)
+	// Special-case time.Time and *time.Time to ensure RFC3339Nano formatting.
+	switch val := v.(type) {
+	case time.Time:
+		return val.UTC().Format(time.RFC3339Nano)
+	case *time.Time:
+		if val == nil {
+			return ""
+		}
+		return val.UTC().Format(time.RFC3339Nano)
```

Comment:
Could unwrap the pointer to avoid defining the format twice

```suggestion
	case *time.Time:
		if val == nil {
			return ""
		}
		return parameterToString(*val, s)
```

Reply:
It makes the code hard to read. I think I would like to skip doing this